WP_Nav update axis, units and comments

lthall opened the pull request:
This is the continued work to update axis and units.

```
SCB: HACK: Removing (build/CubeOrange/modules/DroneCAN/libcanard/dsdlc_generated)
SCB: Running (./waf configure --board CubeOrange --consistent-builds --bootloader) in (.)
SCB: Running (./waf bootloader) in (.)
SCB: Running (rsync -ap build/ /tmp/tmpl7cvn4cm/out-branch-CubeOrange) in (.)
Board,blimp,bootloader,copter,heli,plane,rover,sub
CubeOrange,*,*,*,*,*,*,*
```

peterbarker commented on ArduCopter/mode_auto.cpp:
```diff
@@ -857,23 +857,23 @@ uint32_t ModeAuto::wp_distance() const
     case SubMode::CIRCLE_MOVE_TO_EDGE:
     default:
-        return wp_nav->get_wp_bearing_to_destination();
+        return wp_nav->get_wp_bearing_to_destination_cdeg();
```
We see to have both `cd` and `cdeg` - should we choose one?

rmackay9 replied:
we decided that it should be "cd"

IamPete1 commented on ArduCopter/mode_auto.cpp:
```diff
@@ -614,17 +614,17 @@ void PayloadPlace::start_descent()
     // set vertical speed and acceleration limits
-    pos_control->set_max_speed_accel_U_cm(wp_nav->get_default_speed_down(), wp_nav->get_default_speed_up(), wp_nav->get_accel_z());
-    pos_control->set_correction_speed_accel_U_cmss(wp_nav->get_default_speed_down(), wp_nav->get_default_speed_up(), wp_nav->get_accel_z());
+    pos_control->set_max_speed_accel_U_cm(wp_nav->get_default_speed_down_cms(), wp_nav->get_default_speed_up_cms(), wp_nav->get_accel_U_cmss());
```
Using both `up` and `U` is a little confusing.

lthall replied:
I don't think it is too bad here. Up and Down are explicitly the maximum speeds in those directions where "U" is the axis.

As you say, in the longer term I would like to have a set of defaults at the top of each libary file that says the defaults are NED and m, radians. All variables using alternatives will be labled _deg for example.

I am leaning towards radians being the default internal unit for angle and degrees being the default parameter unit.

lthall replied:
My impression is that People were happy with this on the dev call.

IamPete1 reviewed (commented):
I guess the goal would be to get back to names without units once we get back to SI? This global renaming allows us to make that conversion bit by bit one function at a time.

peterbarker commented:
> I guess the goal would be to get back to names without units once we get back to SI? This global renaming allows us to make that conversion bit by bit one function at a time.

The concept is `void my_function(x_cm);` -> `void my_function_cm(x_cm)` -> `void my_function_cm(x_cm); void my_function_m(x);` -> `void my_function_m(x);`.  Several years down the track we might be able to remove `_m` from the name, but can't until screwing over rebased out-of-tree code is going to be an anomaly rather than a certainty...

rmackay9 commented on ArduCopter/Copter.cpp:
```diff
@@ -474,13 +474,13 @@ AP_Vehicle::custom_mode_state* Copter::register_custom_mode(const uint8_t num, c
     return true;
 }
 
 bool Copter::set_circle_rate(float rate_dps)
```
maybe we should rename Copter::set_circle_rate to Copter::set_circle_rate_degs

rmackay9 replied:
this "rate_dps" should be "rate_degs"

lthall commented on ArduCopter/mode_guided.cpp:
```diff
@@ -1132,7 +1132,7 @@ int32_t ModeGuided::wp_bearing() const
     switch(guided_mode) {
     case SubMode::WP:
-        return wp_nav->get_wp_bearing_to_destination();
+        return wp_nav->get_wp_bearing_to_destination_cdeg();
```
Fixed

lthall commented:
I think I have addressed all comments:

SCB: Running (rsync -ap build/ /tmp/tmp4x269y57/out-branch-CubeOrange) in (.)
Board,blimp,bootloader,copter,heli,plane,rover,sub
CubeOrange,*,*,*,*,*,*,*

peterbarker commented:
Merged, thanks!
